serde-oracle-jdbc-json/src/main/java/io/micronaut/serde/oracle/jdbc/json/OracleJdbcJsonParserDecoder.java

```diff
@@ -102,13 +103,27 @@ protected String coerceScalarToString() {
             case VALUE_DECIMAL:
             case VALUE_DOUBLE:
             case VALUE_FLOAT:
-                // only allowed for string and number
+            case VALUE_INTERVALDS:
+            case VALUE_INTERVALYM:
+            case VALUE_TIMESTAMPTZ:
+            case VALUE_DATE:
+                // only allowed for string, number
+                // additionally for processing string values from VALUE_INTERVALDS, VALUE_INTERVALYM, VALUE_TIMESTAMP,
+                // VALUE_TIMESTAMPTZ and VALUE_DATE
+                // in combination with custom de/serializers configured for Oracle Json parsing
                 return jsonParser.getString();
+            case VALUE_TIMESTAMP:
+                return jsonParser.getLocalDateTime().toString();
             case VALUE_TRUE:
                 return StringUtils.TRUE;
             case VALUE_FALSE:
                 return StringUtils.FALSE;
-            default:
+            case VALUE_BINARY:
+                // This is used to parse metadata _etag from Oracle Json View which is treated as binary value
+                // and getString() returns what we need.
+                OracleJsonBinary binary = jsonParser.getValue().asJsonBinary();
```

Comment:
This does not return consistent results from Oracle JSON. For example, etag field is binary type and it seems is is already encoded using base16. This code works ok for etag but for memo, blob etc it returns base16 encoded strings. If we want to decode, I would need to use google guava library and don't know if that's allowed.

Reply:
Is it specific to the custom value? Maybe we can check for the property name?

Reply:
I have exposed `decodeBinary` method in the decoder and now it can be done easily and more efficient (no need to convert to string etc.)